File: LC-345. Reverse Vowels of a String.cpp
class Solution {
public:
    string reverseVowels(string s) {
     
    int end=s.size()-1;
    int start=0;
    while(start<=end)
    {
           if(s[start]=='A' || s[start]=='E' || s[start]=='I' || s[start]=='O' || s[start]=='U'|| s[start]=='a' || s[start]=='e' || s[start]=='i' || s[start]=='o'|| s[start]=='u' )

        {
            if(s[end]=='a' || s[end]=='e' || s[end]=='i' || s[end]=='o'|| s[end]=='u'||s[end]=='A' || s[end]=='E' || s[end]=='I' || s[end]=='O'|| s[end]=='U') {
                char temp=s[start];
                s[start]=s[end];
                s[end]=temp;
                end--; start++;
            }
            else end--;
        }

        else 
        {
            if (s[end]=='a' || s[end]=='e' || s[end]=='i' || s[end]=='o'|| s[end]=='u'|s[end]=='A' || s[end]=='E' || s[end]=='I' || s[end]=='O'|| s[end]=='U')
            {
                start++;
            }
            else end--,start++;
        }
    }
    return s;
    }
};

File: LC-345. Reverse Vowels of a String.java
class Solution {
    public String reverseVowels(String str) {
        StringBuilder s=new StringBuilder();
        int n=str.length();
        for(int i=0;i<n;i++)
        {
            if(str.charAt(i)=='a' ||str.charAt(i)=='e' ||str.charAt(i)=='i' ||str.charAt(i)=='o' 
            ||str.charAt(i)=='u' ||str.charAt(i)=='A' ||str.charAt(i)=='E' ||str.charAt(i)=='I' 
            ||str.charAt(i)=='O' ||str.charAt(i)=='U')  
                 s.append(str.charAt(i));
        }
        s.reverse();
        StringBuilder result=new StringBuilder();
        int j=0;
        for(int i=0;i<n;i++)
        {
            if(str.charAt(i)=='a' ||str.charAt(i)=='e' ||str.charAt(i)=='i' ||str.charAt(i)=='o' 
            ||str.charAt(i)=='u' ||str.charAt(i)=='A' ||str.charAt(i)=='E' ||str.charAt(i)=='I' 
            ||str.charAt(i)=='O' ||str.charAt(i)=='U')  
            {
                 result.append(s.charAt(j));
                 j++;
            }
            else result.append(str.charAt(i));
        }
        return result.toString();
    }
}
